kafka-connect-s3/src/main/java/io/confluent/connect/s3/continuum/S3Continuum.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.connect.s3.continuum;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.config.AbstractConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.Properties;
+
+import static io.confluent.kafka.serializers.AbstractKafkaSchemaSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG;
+import static org.apache.kafka.clients.CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG;
+
+public class S3Continuum {
+  private static final Logger log = LoggerFactory.getLogger(S3Continuum.class);
+
+  private Producer<Object, Object> producer;
+  private String topic;
+  private int partition;
+  private ObjectMapper mapper;
+  private Schema valueSchema;
+
+  public S3Continuum(AbstractConfig config) {
+    final S3ContinuumConfigValues continuumConfig = S3ContinuumConfig.parseConfigValues(config);
+
+    if (continuumConfig.isConfigured()) {
+      Properties producerProperties = new Properties();
+      producerProperties.put(BOOTSTRAP_SERVERS_CONFIG, continuumConfig.bootstrapServers);
+      producerProperties.put(SCHEMA_REGISTRY_URL_CONFIG, continuumConfig.schemaRegistryURL);
+      producerProperties.put(KEY_SERIALIZER_CLASS_CONFIG,
+              org.apache.kafka.common.serialization.StringSerializer.class);
+      producerProperties.put(VALUE_SERIALIZER_CLASS_CONFIG,
+              continuumConfig.valueConverter);
+      this.producer = new KafkaProducer<>(producerProperties);
+
+      if (continuumConfig.schemaRegistryURL != "") {
+        String s3NotificationSchema =
+                "{\"type\":\"record\","
+                        + "\"name\":\"" + "new_file_ingested_schema" + "_continuum\","
```

Comment:
Is the `+ "new_file_ingestion_schema" +` part there because it was originally a configuration instead of hard-coded? I'm also not sure about how unique the name needs to be. I guess it only has to be unique within the given namespace?

Reply:
Yeah, it originally had the name of the topic. When I was trying to follow [this article](https://medium.com/@igorvlahek1/no-need-for-schema-registry-in-your-spring-kafka-tests-a5b81468a0e1) to write some Avro integration tests without schema registry, I discovered that the name field here isn't allowed to have hyphens. That caused me to want to hardcode the schema name. I wasn't able to get the tests to deserialize the Avro data after following that article, so I kept the tests using JSON, but I was able to verify that the sink could publish Avro when hooked up to the DFS local dev setup.

I spent a little while trying to figure out the naming rules. I think what we're doing here is registering a version of a schema. Schemas get registered under a subject, and it's important for the subject name to be unique. We don't generate the subject name ourselves. Instead, we use the default "TopicNameStrategy," where [the framework generates the name of the subject by appending "-value" to the topic name](https://docs.confluent.io/platform/current/schema-registry/serdes-develop/index.html#subject-name-strategy). Here's what it looks like in Kafka UI:

![image](https://user-images.githubusercontent.com/4745977/202063091-b4cddd52-7d09-4c0a-914d-e63e03f11df4.png)

I think this should work, but we are going to want some runtime on this feature regardless. For some additional manual testing, I created two different tables locally, and both sinks were able to publish to the topic without issue. We still have just the one version of the schema in Kafka UI, so everything seems to be working as expected.

Reply:
<https://avro.apache.org/docs/1.11.1/specification/#names>

From that, it does sound like namespace + name is all that needs to be unique, so we should be fine to use some fixed name given that the namespace is unique enough.

Reply:
So I guess then my question is should we use a simpler name, such as:

```suggestion
                        + "\"name\":\"new_file\","
```

Reply:
Ahh, the documentation for the "name" field was in the AVRO docs. I had only been looking at the schema registry docs. I've simplified the name.